units/time_series_update_periodic_test.go

```diff
@@ -35,6 +39,136 @@ func tearDownPeriodicTest(env cedar.Environment) error {
 	return errors.WithStack(session.DB(conf.DatabaseName).DropDatabase())
 }
 
+func provisionDb(ctx context.Context, env cedar.Environment, rollups []testResultsAndRollups) {
+	for _, result := range rollups {
+		performanceResult := model.CreatePerformanceResult(*result.info, nil, result.rollups)
+		performanceResult.CreatedAt = time.Now().Add(time.Second * -1)
+		performanceResult.Setup(env)
+		err := performanceResult.SaveNew(ctx)
+		if err != nil {
+			panic(err)
+		}
+	}
+}
+
+type testResultsAndRollups struct {
+	info    *model.PerformanceResultInfo
+	rollups []model.PerfRollupValue
+}
+
+func generateDistinctRandoms(existing []int, min, max, num int) []int {
+	var newVals []int
+	exists := map[int]bool{}
+	for _, v := range existing {
+		exists[v] = true
+	}
+	randoms := rand.Perm(max - min + 1)
+	for _, v := range randoms {
+		v += min
+		if !exists[v] {
+			exists[v] = true
+			newVals = append(newVals, v)
+			if len(newVals) == num {
+				break
+			}
+		}
+	}
+	return newVals
+}
+
+func makePerfResultsWithChangePoints(unique string, seed int64) ([]testResultsAndRollups, [][]int) {
+	// Deterministic testing on failure.
+	rand.Seed(seed)
```

Comment:
I guess not, I didn't write these tests so I am not sure the logic behind this move. I also do not think it's worth to do too much refactoring since we are going to move away from this code in the nearish future.